projects/storefrontapp-e2e-cypress/cypress/e2e/accessibility/coupons.a11y-e2e.cy.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as myCoupons from '../../helpers/coupons/my-coupons';
+import { PRODUCT_PATH_2 } from '../../helpers/site-context-selector';
+
+import { viewportContext } from '../../helpers/viewport-context';
+
+describe('Coupons Accessibility', { testIsolation: false }, () => {
+  before(() => {
+    cy.a11yContinuumSetup();
+  });
+
+  viewportContext(['mobile', 'desktop'], () => {
+    describe('My Coupons', () => {
+      before(() => {
+        cy.window().then((win) => {
+          win.sessionStorage.clear();
+        });
+        cy.requireLoggedIn();
+      });
+
+      it('Empty page', () => {
+        cy.visit('/my-account/coupons');
+        cy.findByText(/no coupons/);
+        cy.get('main').a11yRunContinuumTest();
+      });
+
+      it('Claim coupon dialog', () => {
+        cy.visit(myCoupons.myCouponsUrl + '#' + myCoupons.validCouponCode);
+        cy.get('cx-claim-dialog').a11yRunContinuumTest();
+      });
+
+      it('Page with coupons', () => {
+        cy.visit(myCoupons.myCouponsUrl + '#' + myCoupons.validCouponCode);
+        cy.get(
+          '.cx-dialog-body .cx-dialog-row-submit-button .btn:first'
+        ).click();
+        cy.findByText(/customerCoupon1/);
+        cy.get('main').a11yRunContinuumTest();
+      });
+
+      it('Coupon details', () => {
+        cy.get('.cx-card-read-more:first').click();
+        cy.get('cx-coupon-dialog').a11yRunContinuumTest();
+      });
+    });
+
+    describe('Cart Coupon', () => {
+      before(() => {
+        cy.visit(PRODUCT_PATH_2);
+        cy.get('cx-add-to-cart button[type=submit]').click();
+        cy.get('cx-added-to-cart-dialog .cx-modal-content cx-cart-item');
```

Comment:
```suggestion
    const auth = JSON.parse(localStorage.getItem('spartacus⚿⚿auth'));
    cy.addToCart('779841', 1, auth.token.access_token)
```

Instead of going through the UI we can quickly add a product like this.